j5/boards/arduino/uno.py

```diff
@@ -0,0 +1,114 @@
+"""Classes for the Arduino Uno."""
+from enum import IntEnum
+from typing import Mapping, Optional, Set, Type, Union, cast
+
+from j5.backends import Backend
+from j5.boards import Board
+from j5.components import (
+    LED,
+    Component,
+    GPIOPin,
+    GPIOPinInterface,
+    GPIOPinMode,
+    LEDInterface,
+)
+
+
+class AnaloguePin(IntEnum):
+    """Analogue Pins numbering."""
+
+    A0 = 14
+    A1 = 15
+    A2 = 16
+    A3 = 17
+    A4 = 18
+    A5 = 19
+
+
+PinNumber = Union[int, AnaloguePin]
+
+
+class ArduinoUnoBoard(Board):
+    """Arduino Uno Board."""
+
+    _led: LED
+    _digital_pins: Mapping[int, GPIOPin]
+    _analogue_pins: Mapping[AnaloguePin, GPIOPin]
+
+    def __init__(self, serial: str, backend: Backend):
+        self._serial = serial
+        self._backend = backend
+
+        self.setup_arduino()
```

Comment:
If we're calling this upon instantiation should it be marked as private, or is it intended to be called multiple times?

Reply:
This is intended to be called by classes which inherit from this one, (e.g `ServoAssembly`, `Ruggeduino`), but it probably can be private for this purpose anyway.

Reply:
Would `super().__init__(serial, backend)` not be a more Pythonic solution?

Reply:
Specifically:

```python
class ArduinoUnoBoard(Board):
  def __init__(self, serial, backend):
    self._serial = serial
    self._backend = backend
    # generic arduino initialisation goes here
class ServoAssembly(ArduinoUnoBoard):
  def __init__(self, serial, backend):
    super().__init__(serial, backend)
    # servo assembly-specific initialisation goes here
```

Reply:
8a0bf73